### DISCUSSION TITLE
Boolean operation after refine in JS?

### DISCUSSION THREAD

**Discussion by below-1** [elalish/manifold#547] [Q&A] [answered]
After we run refine, Is there a way to prevent triangle collapse in subsequent boolean operation?

**Comment by pca006132** [comment]
I don't think it is possible for now, at least there is no simple way for doing so. One possible way is to assign the faces different mesh ID, as we will not collapse triangles with different mesh ID.

**Comment by elalish** [comment, marked as answer]
The simplest way is to use something like `refined.setProperties(1, (prop) => {prop[0] = Math.random();});`, which by adding a random property value to each vertex, means that generally no two adjacent triangles will be considered coplanar (and therefore collapsable) since that is a function of both geometry and properties.
